packages/actor-dereference-file/lib/ActorDereferenceFile.ts

```diff
@@ -22,13 +22,17 @@ export class ActorDereferenceFile extends ActorDereference {
   }
 
   public async run({ url }: IActionDereference): Promise<IActorDereferenceOutput> {
+    // E.g., sparql@http://example.org/
+    const OptionallyTaggedURIRegex = /(^\w+@)?\w[\w+.-]*:.*/u;
+    const isURI = (str: string): boolean => OptionallyTaggedURIRegex.exec(str) !== null;
+
     const requestTimeStart = Date.now();
     return {
       data: createReadStream(getPath(url)),
       // This should always be after the creation of the read stream
       requestTime: Date.now() - requestTimeStart,
       exists: true,
-      url,
+      url: isURI(url) ? url : pathToFileURL(url).href,
```

Comment:
Aha, I didn't know about this function!
I'm wondering now if all that we need is not just `pathToFileURL`?
We may not even need the `isURI`, since `pathToFileURL` may not do anything if the `file:` scheme is already present.

Reply:
I had hopes it works this way too! To my surprise it cares little whether it's already a URI or not.. seems it considers a file URI to be a file name, prepending it with `file://absolute-path/`, so we something like `file://absolute-path/file://absolute-path`
If this function becomes of frequent use we could create our own helper fn that works the way we want